### PR TITLE
[FRAME] Introduce `force_adjust_total_issuance`

### PR DESCRIPTION
**Pull request by ggwpez**
Add `Balances::force_adjust_total_issuance` as preparation for fixing https://github.com/polkadot-fellows/runtimes/issues/147.  
Important changes in `substrate/frame/balances/src/lib.rs`.

TODO:
- [x] Update weights

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4971338

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4971340

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4971339

**Comment by ggwpez** [comment]
bot bench substrate-pallet --pallet=pallet-balances  
bot bench polkadot-pallet --pallet=pallet-balances --runtime=rococo

**Comment by ggwpez** [comment]
bot bench substrate-pallet --pallet=pallet_balances  
bot bench polkadot-pallet --pallet=pallet_balances --runtime=rococo

File: polkadot/runtime/rococo/src/weights/pallet_balances.rs [hunk 1]
```diff
@@ -93,6 +93,10 @@ impl<T: frame_system::Config> pallet_balances::WeightInfo for WeightInfo<T> {
 			.saturating_add(T::DbWeight::get().reads(1 as u64))
 			.saturating_add(T::DbWeight::get().writes(1 as u64))
 	}
+	
+	// FAIL-CI replace  
+	fn force_adjust_total_issuance() -> Weight { Weight::zero() }
```
Comment (ggwpez) [inline polkadot/runtime/rococo/src/weights/pallet_balances.rs:98]: @mordamax it looks like the bench bot does not yet support the people chain, i think?

**Comment by ggwpez** [comment]
bot bench cumulus-collectives --pallet=pallet_balances  
bot bench cumulus-bridge-hubs --runtime=bridge-hub-rococo --pallet=pallet_balances  
bot bench cumulus-bridge-hubs --runtime=bridge-hub-westend --pallet=pallet_balances  
bot bench cumulus-assets --runtime=asset-hub-westend --pallet=pallet_balances  
bot bench cumulus-assets --runtime=asset-hub-rococo --pallet=pallet_balances   
bot bench cumulus-coretime --runtime=coretime-rococo --pallet=pallet_balances
bot bench cumulus-coretime --runtime=coretime-westend --pallet=pallet_balances  
bot bench polkadot-pallet --runtime=westend --pallet=pallet_balances

**Review by liamaharon — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

File: prdoc/pr_3001.prdoc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+title: Introduce `Balances::force_adjust_total_issuance`
```
Comment (bkchr) [inline prdoc/pr_3001.prdoc:1]:
```suggestion
title: "Introduce `Balances::force_adjust_total_issuance`"
```

**Pull request merged**